src/db.c

```diff
@@ -801,7 +816,27 @@ void keysCommand(client *c) {
     allkeys = (pattern[0] == '*' && plen == 1);
     if (server.cluster_enabled && !allkeys) {
         pslot = patternHashSlot(pattern, plen);
+    } else if (!server.cluster_enabled) {
+        pslot = 0;
```

Comment:
This change affects another `if (pslot != -1) {` below, with kvstoreIteratorInit (for all slots) vs kvstoreGetDictSafeIterator (for slot 0). For non-cluster, it doesn't matter which one we use?

Reply:
yes, in non-cluster mode, we only have the slot 0.